### DISCUSSION TITLE
Effect list

### DISCUSSION THREAD

**Discussion by yisar** [frejs/fre#231] [Ideas]
The new diff algorithm uses two ends traversal, and its order is different from that of fiber. We need to introduce a second linked list.

Such as this:

```js
[ 1, 2, 3, 4 ]
[ 4, 3, 2, 1 ]
```
The order of traversal at both ends is, [ 1, 4, 2, 3 ], because this algorithm is derived from the other frameworks, the order of the effectlist should be the same as that of the traversal at both ends.

Another advantage of effectlist is that it can solve the problem of memory explosion.

Alternative: simplify the diff algorithm and reduce the mobility

**Comment by hkjpotato** [comment]
I didn't notice you already start an effect list algorithm discussion. I will keep updating my finding here and there https://github.com/yisar/fre/discussions/259

**Reply by yisar** [reply]
Fre doesn't use effectlist now, and may use it in the future.

**Reply by hkjpotato** [reply]
Thank you. I am still trying to understand React's effect list, from what I heard it is like a linkedlist built during the construction of the fiber tree but I need to recreate it myself to fully understand it. 

progress tracking here: https://github.com/hkjpotato/tiny-react

should have more free time once I join the open source team next week haah
